packages/form-core/test-suites/FormatMixin.suite.js

```diff
@@ -401,6 +410,38 @@ export function runFormatMixinSuite(customConfig) {
 
         expect(el.formattedValue).to.equal(`foo: ${generatedModelValue}`);
       });
+
+      it('will block the user from inputting invalid values when using a preprocessor', async () => {
+        const preprocessorSpy = sinon.spy(value => value.replace(/[o]/g, ''));
+        const val = generateValueBasedOnType({ viewValue: true }) || 'init-value';
+
+        // Create a copy and run the same preprocessing on it.
+        // So we can check later that the element preprocessor runs
+        // and gives the same output.
+        let processedVal = val;
+        if (typeof val === 'string') {
+          processedVal = val.replace(/[o]/g, '');
+        }
+
+        const el = /** @type {FormatClass} */ (await fixture(html`
+          <${elem} .preprocessor=${preprocessorSpy}>
+            <input slot="input" value="${val}">
+          </${elem}>
+        `));
+
+        if (cfg.tagString?.startsWith('lion-input-date')) {
```

Comment:
Would fail if the suite would be run for an extension

Reply:
Hm thats true, I wonder what a good way to do this is in the suite here.. Any ideas?

Reply:
Maybe check for `modelValue instanceof Date`?
Or use the test below that focuses on viewValues and the fact that the preprocessed value is passed on to the parser (which in turn determines whether modelValue will be invalid or not, depending on its implementation)